web/packages/extension/src/content.ts

```diff
@@ -158,30 +107,14 @@ function isXMLDocument(): boolean {
         return;
     }
 
-    // We must run the plugin polyfill before any flash detection scripts.
-    // Unfortunately, this might still be too late for some websites (issue #969).
-    // NOTE: The script code injected here is the compiled form of
-    // plugin-polyfill.ts. It is injected by tools/inject_plugin_polyfill.ts
-    // which just search-and-replaces for this particular string.
-    // On browsers which support ExecutionWorld MAIN this will be done earlier.
-    if (
-        navigator.wrappedJSObject &&
-        navigator.wrappedJSObject.plugins.namedItem("Shockwave Flash")
-            ?.filename !== "ruffle.js"
-    ) {
-        injectScriptRaw("%PLUGIN_POLYFILL_SOURCE%");
-    }
-
-    await injectScriptURL(utils.runtime.getURL(`dist/ruffle.js?id=${ID}`));
-
     window.addEventListener("message", (event) => {
         // We only accept messages from ourselves.
         if (event.source !== window || !event.data) {
             return;
         }
 
-        const { to, index, data } = event.data;
-        if (to === `ruffle_content${ID}`) {
+        const { to, index, data, id } = event.data;
+        if (to === "ruffle_content" && id === ID) {
```

Comment:
Only look at messages with the same id as was generated (fulfills the same purpose as when previously we were checking only messages to `ruffle_content${ID}`)